master/internal/api_command.go

```diff
@@ -148,6 +148,13 @@ func (a *apiServer) getCommandLaunchParams(ctx context.Context, req *protoComman
 	}
 	taskSpec.UserSessionToken = token
 
+	// ensure that workspace name is propagated to task spec when specified via CLI
+	if req.WorkspaceID != 0 {
+		if w, _ := a.GetWorkspaceByID(ctx, req.WorkspaceID, *userModel, true); w != nil {
+			taskSpec.Workspace = w.Name
+		}
+	}
+
```

Comment:
When you submit a command without a workspace then the Workspace will be empty since this isn't accounting for the defaulting that happens earlier? Commands should always have a workspace name on them since they will always be running in a workspace.

Also can we move this code closer to the workspace setting logic on line 68